Crashes on launch for lack of a client id in tba.properties

To reproduce:
  * Sync'd to master at 19acc7d.
  * Do the Firebase auth setup and add a `./android/src/debug/google-services.json` file.
  * Launch on Android M or N emulator.

Result: Crash, "Unable to start activity ComponentInfo", "Caused by: java.lang.IllegalArgumentException: Given String is empty or null" at `GoogleSignInOptions gso ... .requestIdToken(clientId)` in `GoogleAuthProvider.<init>`.

[Scroll horizontally to read this -------------->
```
08-07 12:30:04.018 2939-2939/com.thebluealliance.androidclient.development W/System.err: java.io.FileNotFoundException: tba.properties

...

                                                                                             --------- beginning of crash
08-07 12:30:04.020 2939-2939/com.thebluealliance.androidclient.development E/AndroidRuntime: FATAL EXCEPTION: main
                                                                                             Process: com.thebluealliance.androidclient.development, PID: 2939
                                                                                             java.lang.RuntimeException: Unable to start activity ComponentInfo{com.thebluealliance.androidclient.development/com.thebluealliance.androidclient.activities.OnboardingActivity}: java.lang.IllegalArgumentException: Given String is empty or null
                                                                                                 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
                                                                                                 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
                                                                                                 at android.app.ActivityThread.-wrap11(ActivityThread.java)
                                                                                                 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
                                                                                                 at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                                 at android.os.Looper.loop(Looper.java:148)
                                                                                                 at android.app.ActivityThread.main(ActivityThread.java:5417)
                                                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
                                                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
                                                                                              Caused by: java.lang.IllegalArgumentException: Given String is empty or null
                                                                                                 at com.google.android.gms.common.internal.zzab.zzhs(Unknown Source)
                                                                                                 at com.google.android.gms.auth.api.signin.GoogleSignInOptions$Builder.zzfs(Unknown Source)
                                                                                                 at com.google.android.gms.auth.api.signin.GoogleSignInOptions$Builder.requestIdToken(Unknown Source)
                                                                                                 at com.thebluealliance.androidclient.auth.google.GoogleAuthProvider.<init>(GoogleAuthProvider.java:42)
                                                                                                 at com.thebluealliance.androidclient.auth.AuthModule.provideGoogleAuthProvider(AuthModule.java:34)
                                                                                                 at com.thebluealliance.androidclient.auth.AuthModule_ProvideGoogleAuthProviderFactory.get(AuthModule_ProvideGoogleAuthProviderFactory.java:31)
                                                                                                 at com.thebluealliance.androidclient.auth.AuthModule_ProvideGoogleAuthProviderFactory.get(AuthModule_ProvideGoogleAuthProviderFactory.java:10)
                                                                                                 at com.thebluealliance.androidclient.auth.AuthModule_ProvideFirebaseAuthProviderFactory.get(AuthModule_ProvideFirebaseAuthProviderFactory.java:36)
                                                                                                 at com.thebluealliance.androidclient.auth.AuthModule_ProvideFirebaseAuthProviderFactory.get(AuthModule_ProvideFirebaseAuthProviderFactory.java:10)
                                                                                                 at com.thebluealliance.androidclient.activities.OnboardingActivity_MembersInjector.injectMembers(OnboardingActivity_MembersInjector.java:39)
                                                                                                 at com.thebluealliance.androidclient.activities.OnboardingActivity_MembersInjector.injectMembers(OnboardingActivity_MembersInjector.java:9)
                                                                                                 at com.thebluealliance.androidclient.di.components.DaggerAuthComponent.inject(DaggerAuthComponent.java:128)
                                                                                                 at com.thebluealliance.androidclient.activities.OnboardingActivity.onCreate(OnboardingActivity.java:90)
                                                                                                 at android.app.Activity.performCreate(Activity.java:6237)
                                                                                                 at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1107)
                                                                                                 at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2369)
                                                                                                 at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476) 
                                                                                                 at android.app.ActivityThread.-wrap11(ActivityThread.java) 
                                                                                                 at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) 
                                                                                                 at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                                                 at android.os.Looper.loop(Looper.java:148) 
                                                                                                 at android.app.ActivityThread.main(ActivityThread.java:5417) 
                                                                                                 at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
                                                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 
```

--- a/android/src/main/java/com/thebluealliance/androidclient/accounts/AccountController.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/accounts/AccountController.java
@@ -47,7 +47,7 @@ public class AccountController {
     }
 
     public boolean isMyTbaEnabled() {
-        return mPreferences.getBoolean(PREF_MYTBA_ENABLED, false);
+        return mPreferences.getBoolean(PREF_MYTBA_ENABLED, false) && !getWebClientId().isEmpty();
     }
 
     public void setSelectedAccount(String account) {

--- a/android/src/main/java/com/thebluealliance/androidclient/auth/google/GoogleAuthProvider.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/auth/google/GoogleAuthProvider.java
@@ -39,8 +39,8 @@ public class GoogleAuthProvider implements AuthProvider,
         String clientId = accountController.getWebClientId();
         if (clientId.isEmpty()) {
             // No client id set in tba.properties, can't continue
-            TbaLogger.w("Oauth client ID not set, can't enable myTBA. See https://goo.gl/Swp5PC " +
-                        "for config details");
+            TbaLogger.w("Oauth client ID not set, can't enable myTBA. See https://goo.gl/Swp5PC "
+                        + "for config details");
             mGoogleApiClient = null;
             return;
         }

--- a/android/src/main/java/com/thebluealliance/androidclient/auth/google/GoogleAuthProvider.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/auth/google/GoogleAuthProvider.java
@@ -8,6 +8,7 @@ import com.google.android.gms.common.api.GoogleApiClient;
 import com.google.android.gms.common.api.OptionalPendingResult;
 import com.google.firebase.auth.AuthCredential;
 
+import com.thebluealliance.androidclient.TbaLogger;
 import com.thebluealliance.androidclient.accounts.AccountController;
 import com.thebluealliance.androidclient.auth.AuthProvider;
 
@@ -17,7 +18,6 @@ import android.os.Bundle;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
 import android.support.annotation.WorkerThread;
-import com.thebluealliance.androidclient.TbaLogger;
 
 import javax.inject.Inject;
 import javax.inject.Singleton;
@@ -30,13 +30,20 @@ public class GoogleAuthProvider implements AuthProvider,
                                            GoogleApiClient.ConnectionCallbacks
 {
 
-    private final GoogleApiClient mGoogleApiClient;
+    private final @Nullable GoogleApiClient mGoogleApiClient;
     private @Nullable GoogleSignInUser mCurrentUser;
 
     @Inject
     public GoogleAuthProvider(Context context, AccountController accountController) {
         mCurrentUser = null;
         String clientId = accountController.getWebClientId();
+        if (clientId.isEmpty()) {
+            // No client id set in tba.properties, can't continue
+            TbaLogger.w("Oauth client ID not set, can't enable myTBA. See https://goo.gl/Swp5PC " +
+                        "for config details");
+            mGoogleApiClient = null;
+            return;
+        }
         GoogleSignInOptions gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
                 .requestEmail()
                 .requestIdToken(clientId)
@@ -54,12 +61,20 @@ public class GoogleAuthProvider implements AuthProvider,
 
     @Override
     public void onStart() {
-        mGoogleApiClient.connect();
+        if (mGoogleApiClient != null
+            && !mGoogleApiClient.isConnecting()
+            && !mGoogleApiClient.isConnected()) {
+            mGoogleApiClient.connect();
+        }
     }
 
     @Override
     public void onStop() {
-        mGoogleApiClient.disconnect();
+        if (mGoogleApiClient != null
+            && !mGoogleApiClient.isConnecting()
+            && mGoogleApiClient.isConnected()) {
+            mGoogleApiClient.disconnect();
+        }
     }
 
     @Override

